Mage.Sets/src/mage/cards/g/GluttonousHellkite.java

```diff
@@ -0,0 +1,173 @@
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.TargetImpl;
+import mage.target.common.TargetSacrifice;
+import mage.util.CardUtil;
+
+/**
+ *
+ * @author grimreap124
+ */
+public final class GluttonousHellkite extends CardImpl {
+
+    public GluttonousHellkite(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{X}{X}{B}{R}{G}");
+        
+        this.subtype.add(SubType.DRAGON);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // When you cast this spell, each player sacrifices X creatures. Gluttonous Hellkite enters the battlefield with two +1/+1 counters on it for each creature sacrificed this way.
+        Ability ability = new CastSourceTriggeredAbility(new GluttonousHellkiteEffect());
+        ability.addEffect(new GluttonousHellkiteReplacementEffect());
+
+        this.addAbility(ability);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+    }
+
+    private GluttonousHellkite(final GluttonousHellkite card) {
+        super(card);
+    }
+
+    @Override
+    public GluttonousHellkite copy() {
+        return new GluttonousHellkite(this);
+    }
+}
+
+class GluttonousHellkiteEffect extends OneShotEffect {
+
+    DynamicValue xValue;
+
+    GluttonousHellkiteEffect() {
+        super(Outcome.Sacrifice);
+        this.staticText = "each player sacrifices X creatures";
+    }
+
+    private GluttonousHellkiteEffect(final GluttonousHellkiteEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public GluttonousHellkiteEffect copy() {
+        return new GluttonousHellkiteEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+
+        List<TargetSacrifice> sacTargets = new ArrayList<>();
+        Player controller = game.getPlayer(source.getControllerId());
+        Object obj = getValue(CastSourceTriggeredAbility.SOURCE_CAST_SPELL_ABILITY);
+        int sacrificeAmount = 0;
+        if (controller == null ) {
+            return false;
+        }
+        if (obj instanceof SpellAbility) {
+            sacrificeAmount = ((SpellAbility) obj).getManaCostsToPay().getX();
+        }
+        if (sacrificeAmount < 1) {
+            return true;
+        }
+        for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
+            Player player = game.getPlayer(playerId);
+            if (player != null) {
+                TargetSacrifice target = new TargetSacrifice(sacrificeAmount, StaticFilters.FILTER_PERMANENT_CREATURE);
+                if (target.canChoose(player.getId(), source, game)) {
+                    player.choose(Outcome.Sacrifice, target, source, game);
+                    sacTargets.add(target);
```

Comment:
why store the TargetSacrifice? wouldn't it make more sense to store a list of the chosen Permanents?

Reply:
How do you do that? I copied this code from another card but I've forgotten which now

Reply:
just instead of a `List<TargetSacrifice>` make a `List<Permanent>`, and use `getTargets()` and `getPermanent()` here instead of in the later loop